### PR TITLE
Moving the AnalyzerEngine to the init to avoid initialization on every request

### PR DESCRIPTION
**Pull request by omri374**
+ some generalization of the classes to allow easier extension and customization.

### CODE REVIEW

File: presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py [hunk 1]
```diff
@@ -1,28 +1,44 @@
...
+    :param analyzer_engine: The Presidio AnalyzerEngine instance
+        to be used to detect PII in text
+    :param ocr: the OCR object to be used to detect text in images.
+    """
+
+    def __init__(self, analyzer_engine: AnalyzerEngine = None, ocr: OCR = None):
+        if not analyzer_engine:
+            self.analyzer_engine = AnalyzerEngine()
+        else:
+            self.analyzer_engine = analyzer_engine
```
Comment (shiranr) [inline presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py:21]:
```suggestion
           if not analyzer_engine:
                   analyzer_engine = AnalyzerEngine()
            self.analyzer_engine = analyzer_engine
```

File: presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+
+        if not ocr:
+            self.ocr = TesseractOCR()
+        else:
+            self.ocr = ocr
```
Comment (shiranr) [inline presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py:26]:
```suggestion
           if not ocr:
               ocr = TesseractOCR()
          self.ocr = ocr
```

**Review by shiranr — COMMENTED** [review]
LGTM, small comments

**Review by navalev — APPROVED** [review]

File: presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py (hunk 1 above)
Comment (omri374) [reply presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py]: Fixed

File: presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py (hunk 2 above)
Comment (omri374) [reply presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py]: Fixed

**Review by navalev — APPROVED** [review]

**Review by shiranr — APPROVED** [review]

**Pull request merged**